### DISCUSSION TITLE
Duckdb timing the sql

### DISCUSSION THREAD

**Discussion by WyattLiu** [duckdb/duckdb#9588] [Q&A] [answered]
May I ask if Duckdb provided some mechanism to issue timers in duckdb similar to chrono? So we can time the execution time for certain statements, even in SQL.

**Comment by l1t1** [comment]
1. CLI .timer
2. explain analyze select  ...

**Reply by WyattLiu** [reply]
Thank you for your reply, May I ask if we have a documentation how to use CLI .timer?

**Reply by Mause** [reply]
DuckDB documentation is available on our website: https://duckdb.org/docs/archive/0.9.1/api/cli#special-commands-dot-commands

**Reply by WyattLiu** [reply]
I am grateful to have some pointers, I tried with .timer on, it seems only time one statement, may I ask what should I do if I want to time a block of statements? I tried to write a procedure, but it seems duckdb does not like it.

I am wondering if we have any CLI way to time from A to B.

Best Regards.

**Reply by WyattLiu** [reply]
I also tried to wrap it in with FUNCTION or PROCEDURE, I wished it could just time the whole block. But it seems duckdb does not support user-defined functions. Or may I ask if I missed anything?
Error: near line 72: Parser Error: syntax error at or near "RETURNS"
LINE 1: CREATE FUNCTION load_data_from_file() RETURNS void AS

CREATE FUNCTION load_data_from_file() RETURNS void AS
BEGIN
    COPY statements...
END;

**Reply by Maxxen** [reply]
DuckDB does not support pl/sql user defined functions. You can however use `;` to separate multiple statements and then execute them all at once in the CLI with `.timer on` to get the total runtime.

**Reply by WyattLiu** [reply]
Thank you!

**Comment by l1t1** [comment, marked as answer]
thanks, Maxxen
```
D .timer on
D select 1;select 2;
┌───────┐
│   1   │
│ int32 │
├───────┤
│     1 │
└───────┘
┌───────┐
│   2   │
│ int32 │
├───────┤
│     2 │
└───────┘
Run Time (s): real 0.003 user 0.003599 sys 0.000000
```

**Comment by WyattLiu** [comment]
Thank you all for commenting on it. If we could have 
.timer symbol_A start
# time-consuming statements 
.timer symbol_A stop
.timer symbol_A report

That would be ideal. But if people are comfortable using python or C++ APIs, they can time in whatever way possible.

Thank you.
